storage/store.go

```diff
@@ -1382,14 +1387,25 @@ func (s *Store) resolveWriteIntentError(ctx context.Context, wiErr *proto.WriteI
 			},
 		}
 		resolveReply := &proto.InternalResolveIntentResponse{}
-		// Add resolve command with wait=false to add to Raft but not wait for completion.
-		waitForResolve := wait && i == len(wiErr.Intents)-1
-		if resolveErr := rng.AddCmd(ctx, proto.Call{Args: resolveArgs, Reply: resolveReply}, waitForResolve); resolveErr != nil {
-			if log.V(1) {
-				log.Warningc(ctx, "resolve for key %s failed: %s", intent.Key, resolveErr)
+
+		go func(errCh chan<- error, key proto.Key) {
```

Comment:
I understand why `key` is an argument to the goroutine, but why `errCh`? It doesn't change so it should be safe to simply close over.